grpclog/component.go

```diff
@@ -0,0 +1,289 @@
+/*
+ *
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package grpclog
+
+import (
+	"fmt"
+	"math"
+	"os"
+	"strconv"
+	"strings"
+
+	"google.golang.org/grpc/internal/grpclog"
+)
+
+// Component is a unique identifier for a component. Created with NewComponent.
+type Component string
```

Comment:
Can we make a slight change here?  Rename `NewComponent` to `Component`:

```
func Component(name string) LoggerV2
```

Then I think we can delete the `Component` type -- maybe `componentData` is all we need, and it could have all the `LoggerV2` methods defined on it.

Let me know if that's not clear or if it doesn't sound like it would work / be a good idea.

Reply:
NewComponent replaced with Component, and Component type was removed and replaced with ComponentData. Problem with making ComponentData a LoggerV2 is that components through ComponentData actually use the current logger to log, so making a component the current logger would create an infinite loop.

Reply:
> Problem with making ComponentData a LoggerV2 is that components through ComponentData actually use the current logger to log, so making a component the current logger would create an infinite loop.

That doesn't sound right.  I meant that we could implement the `LoggerV2` methods, not that it would become the globally-installed logger.

Reply:
Why have it implement the LoggerV2 interface if not to make it an installable LoggerV2 through the grpclog interface?

Reply:
My thinking was that could be the return type from `Component`.